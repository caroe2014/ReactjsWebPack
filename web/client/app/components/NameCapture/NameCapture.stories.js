// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React from 'react';
import { storiesOf } from '@storybook/react';
import NameCapture from '../NameCapture';
import i18n from '../../../i18n';
import { I18nextProvider } from 'react-i18next';

storiesOf('NameCapture', module)
  .addDecorator(story => <I18nextProvider i18n={i18n}>{story()}</I18nextProvider>)
  .add('No nameInstructionText', () => {
    return (
      <NameCapture/>
    );
  }
  )
  .add('nameInstructionText', () => {
    return (
      <NameCapture nameInstructionText='Instruction text'/>
    );
  }
  ).add('without skip', () => {
    return (
      <NameCapture nameInstructionText='Instruction text' isNameCaptureRequired/>
    );
  }
  );
