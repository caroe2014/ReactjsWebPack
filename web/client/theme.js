// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import { css, injectGlobal } from 'styled-components';

const theme = {
  colors: {
    black: '#000000',
    navigationMenu: '#48a942',
    navigationMenuHover: 'cornflowerblue',
    primary: '#347CFF',
    primaryButton: '#347CFF',
    loginField: '#AAA7AC',
    placeHolder: '#909090',
    loginBoxColor: '#434149',
    open: '#48a942',
    close: 'red',
    footer: '#48a942',
    textGrey: '#505050',
    paymentSuccessTitle: '#505050',
    inputBottomBorder: 'lightgray',
    backgroundGrey: '#E6E6E6',
    text: 'black',
    primaryText: '#383838',
    secondaryText: '#a6a6a6',
    hint: '#a6a6a6',
    light: 'white',
    error: '#cc0000',
    tile: '#505050',
    addToCart: '#347CFF',
    itemTile: '#3C474C0',
    darkGrey: '#464646',
    itemBorder: 'lightgrey',
    addOnTab: '#347CFF',
    validationError: '#cc0000',
    loginBackground: '#303030',
    loginFormBackground: '#484848',
    disableBackground: '#E4E4E4',
    addonBg: '#EEEEEE',
    glutenFree: '#8d7356',
    spicy: '#ed1c24',
    vegan: '#0d4a22',
    vegetarian: '#0d4a22',
    new: '#ff6000',
    healthy: '#359919',
    contextBackground: '#f0f0f3',
    loader: '#ababab',
    disableButton: '#D1D1D1'
  },
  fontFamily: 'Roboto',
  receiptFontFamily: 'Roboto',
  fontSize: {
    xl: '2.4rem',
    lg: '1.8rem',
    md: '1.3rem',
    nm: '1rem',
    sm: '0.75rem'
  },
  fontSizeFromProps: (props) => {
    if (props.xl) return props.theme.fontSize.xl;
    if (props.lg) return props.theme.fontSize.lg;
    if (props.md) return props.theme.fontSize.md;
    if (props.nm) return props.theme.fontSize.nm;
    if (props.sm) return props.theme.fontSize.sm;
    return 'inherit';
  },
  spacing: {
    xl: '8em',
    lg: '4em',
    md: '2em',
    nm: '1em',
    sm: '0.5em',
    none: '0em'
  },
  mediaBreakpoints: {
    mobile: (...args) => css`@media (max-width: 32em) {${css(...args)}}`,
    tablet: (...args) => css`@media (max-width: 72em) {${css(...args)}}`,
    desktop: (...args) => css`@media (min-width: 72em) {${css(...args)}}`
  },
  shadow: '#cfcfcf 0.1px 0.1px 3px 0px',
  nav: {
    height: '70px'
  }
};

injectGlobal`
  .agilysys-icon-new {
    color: ${theme.colors.new}
  }
  .agilysys-icon-healthy {
    color: ${theme.colors.healthy}
  }
  .agilysys-icon-vegetarian {
    color: ${theme.colors.vegetarian}
  }
  .agilysys-icon-vegan {
    color: ${theme.colors.vegan}
  }
  .agilysys-icon-spicy {
    color: ${theme.colors.spicy}
  }
  .agilysys-icon-gluten-free {
    color: ${theme.colors.glutenFree}
  }
  @media print {
    @-moz-document url-prefix() {
      .appContainer, .appContainer>div, .appContainer .parent, .appContainer .BottomContainer {
        display: block !important;
      }
    }
  }
`;

export default theme;
