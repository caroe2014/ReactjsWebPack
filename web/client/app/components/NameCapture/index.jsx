// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React, { Component } from 'react';
import styled, { ThemeProvider } from 'styled-components';
import theme from 'web/client/theme';
import { Text, Flex, Button } from 'rebass';
import FloatingLabelInput from 'web/client/app/components/FloatingLabelInput';
import { AVOID_ONLY_FIRST_SPACE, FIRST_NAME_PATTERN, LAST_INITIAL_PATTERN } from 'web/client/app/utils/constants';
import { Trans } from 'react-i18next';
import i18n from 'web/client/i18n';

const Container = styled(Flex)`
  align-items: center;
  margin: auto;
  max-width: 400px;
  padding-bottom: 20px;
  ${props => props.theme.mediaBreakpoints.desktop`margin-top: 170px;`};
  ${props => props.theme.mediaBreakpoints.tablet`margin-top: 170px;`};
  ${props => props.theme.mediaBreakpoints.mobile`
    margin-top: 120px; padding-bottom: 20px; 
    border: none;
    max-width: 100%; 
    height: inherit;
    padding-left: 10px;
    padding-right: 10px;
    background: ${props => props.theme.colors.light};
  `};
  flex-direction: column;
  width: 100%;
  margin-bottom: auto !important;
`;

const ChildBox = styled(Flex)`
  flex-direction: column;
  width: 100%;
  word-break: break-word;
  background: ${props => props.theme.colors.light};
  padding: 20px;
  ${props => props.theme.mediaBreakpoints.desktop`border: 1px solid lightgrey; margin-bottom: 20px`};
  ${props => props.theme.mediaBreakpoints.tablet`border: 1px solid lightgrey;`};
  ${props => props.theme.mediaBreakpoints.mobile`border: none; margin-top: 0px; padding: 20px 0px`};
`;

const InputContainer = styled(Flex)`
  flex-direction: column;
  border-bottom: none;
  border-top: none;
  padding-top: 20px;
  width: 100%;
  align-self: center;
  margin-top: 20px;
`;

const HeaderLabel = styled(Text)`
  color: ${props => props.theme.colors.primaryTextColor};
  font-size: ${props => props.theme.fontSize.md};
  font-weight: 500;
  max-width: 360px;
  justify-content: left;
  text-align: center;
  ${props => props.theme.mediaBreakpoints.mobile`
    max-width: 100%;
    padding-right: 10px;
  `};
  &:focus {
    outline: none;
    box-shadow: none;
  }
`;

const ButtonParent = styled(Flex)`
  flex-direction: column;
  min-height: 60px;
`;

const PayButton = styled(Button)`
  background: ${props => props.theme.colors.buttonControlColor};
  color: ${props => props.theme.colors.buttonTextColor};
  cursor: pointer;
  font-weight: 500;
  font-size: ${props => props.theme.fontSize.md};
  float: right;
  width: 100%;
  height: 60px;
  min-height: 60px;
  padding: ${props => props.theme.spacing.sm} ${props => props.theme.spacing.nm};
  &:hover {
    box-shadow: 0px 0px 10px 2px lightgrey;
    -webkit-box-shadow: 0px 0px 10px 2px lightgrey;
    -moz-box-shadow: 0px 0px 10px 2px lightgrey;;
  }
  text-align: center;
  margin: 0 auto;
  border: none;
  border-radius: 0;
  outline: none;
  &:focus {
    outline: none;
    box-shadow: none;
  }
`;

const SkipButton = styled(Button)`
  background: ${props => props.theme.colors.light};
  border: 0.5px solid ${props => props.theme.colors.buttonControlColor};
  color: ${props => props.theme.colors.buttonControlColor};
  cursor: pointer;
  font-weight: 500;
  font-size: ${props => props.theme.fontSize.md};
  float: right;
  width: 100%;
  height: 60px;
  min-height: 60px;
  padding: ${props => props.theme.spacing.sm} ${props => props.theme.spacing.nm};
  &:hover {
    box-shadow: 0px 0px 10px 2px lightgrey;
    -webkit-box-shadow: 0px 0px 10px 2px lightgrey;
    -moz-box-shadow: 0px 0px 10px 2px lightgrey;
  }
  text-align: center;
  margin: 15px auto 0 auto;
  border-radius: 0;
  outline: none;
  &:focus {
    outline: none;
    box-shadow: none;
  }
`;

const InputHeader = styled(Text)`
  color: ${props => props.theme.colors.primaryTextColor};
  font-size: ${props => props.theme.fontSize.nm};
  margin-bottom: 10px;
  text-align: center;
`;

class NameCapture extends Component {

  constructor (props) {
    super(props);
    this.firstNameProperty = 'firstName';
    this.lastInitialProperty = 'lastInitial';
    this.state = {firstNameError: '',
      lastInitialError: '',
      firstName: this.props.firstName || '',
      lastInitial: this.props.lastInitial || ''};
    this.skipNameCapture = this.skipNameCapture.bind(this);
    this.continuePay = this.continuePay.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }

  componentDidMount () {
    window.scrollTo(0, 0);
  }

  handleChange (propertyName, inputValue) {
    let errorData = {};
    if (propertyName === this.firstNameProperty) {
      errorData.firstNameError = inputValue ? '' : this.state.firstNameError;
    } else if (propertyName === this.lastInitialProperty) {
      errorData.lastInitialError = inputValue ? '' : this.state.lastInitialError;
    }
    this.setState({[propertyName]: inputValue, ...errorData});
  }

  continuePay () {
    const { firstName, lastInitial } = this.state;
    const { onContinuePay, getAliasName } = this.props;
    this.setState({
      firstNameError: firstName.length === 0 ? i18n.t(getAliasName ? 'ALIAS_NAME_ERROR' : 'FIRST_NAME_ERROR') : '',
      lastInitialError: (!getAliasName && lastInitial.length === 0) ? i18n.t('LAST_INITIAL_ERROR') : ''});
    if (firstName.length === 0 || (!getAliasName && lastInitial.length === 0)) {
      return;
    }
    onContinuePay(this.state.firstName, this.state.lastInitial);
  }

  skipNameCapture () {
    const { onContinuePay } = this.props;
    this.setState({firstNameError: '',
      lastInitialError: ''});
    onContinuePay('', '');
  }

  render () {
    const { isNameCaptureRequired, nameInstructionText, isSmsEnabled, deliveryEnabled, getAliasName } = this.props;
    const { firstNameError, lastInitialError, firstName, lastInitial } = this.state;
    return (
      <ThemeProvider theme={theme}>
        <Container className='name-capture-container'>
          <ChildBox className='child-box'>
            <HeaderLabel
              aria-label={nameInstructionText || i18n.t(getAliasName
                ? 'ALIAS_NAME_CAPTURE_INSTN' : 'NAME_CAPTURE_INSTN')}
              tabIndex={0}
              role='form' className='instruction-text'>
              {nameInstructionText || <Trans i18nKey={getAliasName
                ? 'ALIAS_NAME_CAPTURE_INSTN' : 'NAME_CAPTURE_INSTN'}/>}
            </HeaderLabel>
            <InputContainer>
              {(isNameCaptureRequired || getAliasName) &&
                <InputHeader tabIndex={0} aria-label={i18n.t('DELIVERY_LOCATION_INPUT_HEADER')}>
                  <Trans i18nKey='DELIVERY_LOCATION_INPUT_HEADER'/></InputHeader>
              }
              <FloatingLabelInput
                ariaLabel={i18n.t(getAliasName ? 'ALIAS_NAME_LABEL' : 'FIRST_NAME_LABEL')}
                tabIndex={0}
                propertyName={this.firstNameProperty}
                label={<Trans i18nKey={getAliasName ? 'ALIAS_NAME_LABEL' : 'FIRST_NAME_LABEL'}/>}
                value={firstName}
                error={firstNameError}
                callBack={this.handleChange}
                clearIcon
                validationRegEx={getAliasName ? AVOID_ONLY_FIRST_SPACE : FIRST_NAME_PATTERN}
                placeHolder={<Trans i18nKey={getAliasName ? 'ALIAS_NAME_LABEL' : 'FIRST_NAME_LABEL'}/>}/>
              {!getAliasName &&
              <FloatingLabelInput
                ariaLabel={i18n.t('LAST_INITIAL_LABEL')}
                tabIndex={0}
                className='lastInitial'
                propertyName={this.lastInitialProperty}
                label={<Trans i18nKey='LAST_INITIAL_LABEL'/>}
                value={lastInitial}
                error={lastInitialError}
                callBack={this.handleChange}
                clearIcon
                validationRegEx={LAST_INITIAL_PATTERN}
                placeHolder={<Trans i18nKey='LAST_INITIAL_LABEL'/>}/>
              }
              <ButtonParent>
                <PayButton
                  aria-label={isSmsEnabled || deliveryEnabled ? i18n.t('NEXT_LABEL')
                    : i18n.t('DELIVERY_LOCATION_CONTINUE_PAY')}
                  tabIndex={0}
                  className='nextBtn'
                  children={isSmsEnabled || deliveryEnabled ? <Trans i18nKey='NEXT_LABEL'/>
                    : <Trans i18nKey='DELIVERY_LOCATION_CONTINUE_PAY'/>}
                  onClick={this.continuePay}/>
                {!getAliasName && !isNameCaptureRequired && <SkipButton
                  aria-label={i18n.t('SKIP_LABEL')}
                  tabIndex={0}
                  className='skip-btn'
                  children={<Trans i18nKey='SKIP_LABEL'/>}
                  onClick={this.skipNameCapture}/>}
              </ButtonParent>
            </InputContainer>
          </ChildBox>
        </Container>
      </ThemeProvider>
    );
  }
}

export default NameCapture;
