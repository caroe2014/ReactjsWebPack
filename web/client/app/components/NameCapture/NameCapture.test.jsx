// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React from 'react';
import { shallow, mount, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import NameCapture from '.';
import jsdom from 'jsdom';
import { ThemeProvider } from 'styled-components';
import theme from 'web/client/theme';

configure({
  adapter: new Adapter()
});

/* global describe, it, expect, beforeEach jest */
/* eslint-disable max-len */
global.window = new jsdom.JSDOM().window;
global.document = window.document;
global.scrollTo = jest.fn();

jest.mock('web/client/i18n', () => ({
  t: (key) => {
    return key;
  }
}));
jest.mock('react-i18next', () => ({
  // this mock makes sure any components using the translate HoC receive the t function as a prop
  withTranslation: () => Component => {
    Component.defaultProps = { ...Component.defaultProps, t: (key) => key };
    return Component;
  },
}));

jest.mock('react-i18next', () => ({
  Trans: 'div'
}));

configure({
  adapter: new Adapter()
});

describe('NameCapture', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<NameCapture isNameCaptureRequired nameInstructionText='Instruction text'/>
    );
  });

  it('does a snapshot test', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render without throwing an error', () => {
    expect(wrapper.exists()).toBe(true);
  });

  it('should render without throwing an error on mount', () => {
    expect(mount(<ThemeProvider theme={theme}><NameCapture isNameCaptureRequired nameInstructionText='Instruction text'/></ThemeProvider>).exists()).toBe(true);
  });

  it('should render without throwing an error on mount without NameCaptureRequired', () => {
    expect(mount(<ThemeProvider theme={theme}><NameCapture isNameCaptureRequired={false} nameInstructionText='Instruction text'/></ThemeProvider>).exists()).toBe(true);
  });

  // it('should render skip button when prop is true', () => {
  //   expect(wrapper.dive().find('.skip-btn').length).toBe(1);
  // });

  describe('when Instruction text empty', () => {
    it('should render Instruction text when smsInstructionText prop not received', () => {
      const TipWrapper = shallow(<NameCapture isNameCaptureRequired={false} isSmsEnabled/>);
      expect(TipWrapper.dive().find('.instruction-text').length).toBe(1);
    });
  });

  describe('when Sms is not Enabled', () => {
    it('should render Instruction text when smsInstructionText prop not received', () => {
      const TipWrapper = shallow(<NameCapture isNameCaptureRequired={false} isSmsEnabled={true}/>);
      expect(TipWrapper.dive().find('.instruction-text').length).toBe(1);
    });
  });

  it('should call skipNameCapture with spy', () => {
    let mockFn = jest.fn();
    let wrapper = shallow(<NameCapture isNameCaptureRequired={false}  onContinuePay={mockFn} isSmsEnabled={true}/>);
    const spy = jest.spyOn(wrapper.instance(), 'skipNameCapture');
    wrapper.instance().forceUpdate();
    wrapper.find('.skip-btn').simulate('click');
    expect(spy).toHaveBeenCalled();
    spy.mockClear();
  });

  describe('when continuePay is clicked', () => {
  it('should call continuePay with spy', () => {
    let mockFn = jest.fn();
    let wrapper = shallow(<NameCapture isNameCaptureRequired={false}  onContinuePay={mockFn} isSmsEnabled={true}/>);
    const spy = jest.spyOn(wrapper.instance(), 'continuePay');
    wrapper.instance().forceUpdate();
    wrapper.find('.nextBtn').simulate('click');
    expect(spy).toHaveBeenCalled();
    spy.mockClear();
  });
  it('firstName, lastInitial is given', () => {
    let mockFn = jest.fn();
    let wrapper = shallow(<NameCapture isNameCaptureRequired={false}  onContinuePay={mockFn} isSmsEnabled={true}/>);
    const spy = jest.spyOn(wrapper.instance(), 'continuePay');
    wrapper.setState({ firstName: 'Test', lastInitial: 'test' });
    wrapper.instance().forceUpdate();
    wrapper.find('.nextBtn').simulate('click');
    expect(spy).toHaveBeenCalled();
    spy.mockClear();
  });
});


});
