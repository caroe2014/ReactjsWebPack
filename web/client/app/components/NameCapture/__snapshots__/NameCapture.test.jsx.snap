// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameCapture does a snapshot test 1`] = `
<ThemeProvider
  theme={
    Object {
      "colors": Object {
        "addOnTab": "#347CFF",
        "addToCart": "#347CFF",
        "addonBg": "#EEEEEE",
        "backgroundGrey": "#E6E6E6",
        "black": "#000000",
        "close": "red",
        "contextBackground": "#f0f0f3",
        "darkGrey": "#464646",
        "disableBackground": "#E4E4E4",
        "disableButton": "#D1D1D1",
        "error": "#cc0000",
        "footer": "#48a942",
        "glutenFree": "#8d7356",
        "healthy": "#359919",
        "hint": "#a6a6a6",
        "inputBottomBorder": "lightgray",
        "itemBorder": "lightgrey",
        "itemTile": "#3C474C0",
        "light": "white",
        "loader": "#ababab",
        "loginBackground": "#303030",
        "loginBoxColor": "#434149",
        "loginField": "#AAA7AC",
        "loginFormBackground": "#484848",
        "navigationMenu": "#48a942",
        "navigationMenuHover": "cornflowerblue",
        "new": "#ff6000",
        "open": "#48a942",
        "paymentSuccessTitle": "#505050",
        "placeHolder": "#909090",
        "primary": "#347CFF",
        "primaryButton": "#347CFF",
        "primaryText": "#383838",
        "secondaryText": "#a6a6a6",
        "spicy": "#ed1c24",
        "text": "black",
        "textGrey": "#505050",
        "tile": "#505050",
        "validationError": "#cc0000",
        "vegan": "#0d4a22",
        "vegetarian": "#0d4a22",
      },
      "fontFamily": "Roboto",
      "fontSize": Object {
        "lg": "1.8rem",
        "md": "1.3rem",
        "nm": "1rem",
        "sm": "0.75rem",
        "xl": "2.4rem",
      },
      "fontSizeFromProps": [Function],
      "mediaBreakpoints": Object {
        "desktop": [Function],
        "mobile": [Function],
        "tablet": [Function],
      },
      "nav": Object {
        "height": "70px",
      },
      "receiptFontFamily": "Roboto",
      "shadow": "#cfcfcf 0.1px 0.1px 3px 0px",
      "spacing": Object {
        "lg": "4em",
        "md": "2em",
        "nm": "1em",
        "none": "0em",
        "sm": "0.5em",
        "xl": "8em",
      },
    }
  }
>
  <Styled(Flex)
    className="name-capture-container"
  >
    <Styled(Flex)
      className="child-box"
    >
      <Styled(Text)
        aria-label="Instruction text"
        className="instruction-text"
        role="form"
        tabIndex={0}
      >
        Instruction text
      </Styled(Text)>
      <Styled(Flex)>
        <Styled(Text)
          aria-label="DELIVERY_LOCATION_INPUT_HEADER"
          tabIndex={0}
        >
          <div
            i18nKey="DELIVERY_LOCATION_INPUT_HEADER"
          />
        </Styled(Text)>
        <FloatingLabelInput
          ariaLabel="FIRST_NAME_LABEL"
          callBack={[Function]}
          clearIcon={true}
          error=""
          label={
            <div
              i18nKey="FIRST_NAME_LABEL"
            />
          }
          placeHolder={
            <div
              i18nKey="FIRST_NAME_LABEL"
            />
          }
          propertyName="firstName"
          tabIndex={0}
          validationRegEx={/\\^\\(\\[a-z\\|A-Z\\]\\[,\\.\\]\\?\\[ \\]\\?\\|\\[a-z\\|A-Z\\]\\['-\\]\\?\\)\\*\\$/}
          value=""
        />
        <FloatingLabelInput
          ariaLabel="LAST_INITIAL_LABEL"
          callBack={[Function]}
          className="lastInitial"
          clearIcon={true}
          error=""
          label={
            <div
              i18nKey="LAST_INITIAL_LABEL"
            />
          }
          placeHolder={
            <div
              i18nKey="LAST_INITIAL_LABEL"
            />
          }
          propertyName="lastInitial"
          tabIndex={0}
          validationRegEx={/\\^\\[a-z\\|A-Z\\]\\[a-z\\|A-Z\\\\s\\]\\{0\\}\\$/}
          value=""
        />
        <Styled(Flex)>
          <Styled(Button)
            aria-label="DELIVERY_LOCATION_CONTINUE_PAY"
            className="nextBtn"
            onClick={[Function]}
            tabIndex={0}
          >
            <div
              i18nKey="DELIVERY_LOCATION_CONTINUE_PAY"
            />
          </Styled(Button)>
        </Styled(Flex)>
      </Styled(Flex)>
    </Styled(Flex)>
  </Styled(Flex)>
</ThemeProvider>
`;
